src/confd/yang/confd/infix-dhcpv6-client.yang

```diff
@@ -0,0 +1,135 @@
+module infix-dhcpv6-client {
+  yang-version 1.1;
+  namespace "urn:infix:params:xml:ns:yang:dhcpv6-client";
+  prefix dhcpv6-client;
+
+  import ietf-interfaces {
+    prefix if;
+  }
+  import ietf-ip {
+    prefix ip;
+  }
+  import infix-dhcp-common {
+    prefix dhcp;
+  }
+
+  contact      "kernelkit@googlegroups.com";
+  description  "This module implements a DHCPv6 client";
+
+  revision 2025-11-09 {
+    description   "Initial revision, split from infix-dhcp-client.
+                      - Separate module for DHCPv6 client
+                      - Augment for /interfaces/interface/ipv6/dhcp
+                      - Support for DHCPv6 options, DUID, prefix delegation";
+    reference     "rfc8415";
+  }
+
+  /*
+   * Typedefs
+   */
+
+  typedef route-preference {
+    type uint32;
+    description "This type is used for selecting route preference (distance).";
+  }
+
+  /*
+   * DHCPv6 Client Configuration
+   */
+
+  augment "/if:interfaces/if:interface/ip:ipv6" {
+    container dhcp {
+      presence "Enable DHCPv6 client for this interface.";
+      description "DHCPv6 client configuration";
+
+      leaf duid {
+        type dhcp:duid;
+        description "Optional DHCPv6 Unique Identifier (DUID), option 1, RFC 8415.
+
+                     When omitted, the client automatically generates a DUID
+                     based on link-layer address plus time (DUID-LLT) or link-layer
+                     address only (DUID-LL), depending on the implementation.
+
+                     This setting allows manual configuration of the DUID in
+                     hexadecimal format. The DUID type and content are encoded
+                     as defined in RFC 8415 section 11.
+
+                     For full control of the DUID format, use the generic option
+                     list instead, which supports entering raw HEX data.
+
+                     Example: 00:03:00:01:08:00:27:fe:8c:2a
+                              (DUID-LL with Ethernet address)";
+      }
+
+      leaf information-only {
+        type boolean;
+        description "Use stateless DHCPv6 (information-request mode).
+
+                     When enabled, the client requests configuration information
+                     (like DNS servers) but does not request IPv6 addresses or
+                     prefixes. This is used with SLAAC (Stateless Address
+                     Autoconfiguration) when only additional configuration is
+                     needed from DHCPv6.";
+      }
```

Comment:
Should there be a `must` expression that requires that `autoconf` is enabled on the interface? What happens otherwise?